lib/dal/src/action/runner.rs

```diff
@@ -444,30 +474,35 @@ impl ActionRunner {
     }
 
     /// Generates a [`ActionHistoryView`] based on [`self`](ActionRunner).
-    pub async fn history_view(&self) -> ActionRunnerResult<Option<ActionHistoryView>> {
-        Ok(Some(ActionHistoryView {
+    pub async fn history_view(&self) -> ActionRunnerResult<ActionHistoryView> {
+        Ok(ActionHistoryView {
             id: self.id,
-            status: if self.resource.is_none() {
-                ActionCompletionStatus::Unstarted
-            } else {
-                self.completion_status
-                    .unwrap_or(ActionCompletionStatus::Failure)
-            },
+            status: self
+                .completion_status
+                .unwrap_or(ActionCompletionStatus::Unstarted),
```

Comment:
This is giving me nightmares from us bug fixing this over the past year or so. Surely it will work this time!